ang/civicase/Utils.js

```diff
@@ -236,39 +236,59 @@
   });
 
   module.factory('getActivityFeedUrl', function ($route, $location, $sce) {
-    return function (caseId, category, statusType, status, id) {
-      caseId = parseInt(caseId, 10);
-      var af = {};
+    return function (urlParams) {
+      var activityFilters = {};
+      var baseUrl = '#/case/list?';
       var currentPath = $location.path();
-      if (category) {
-        af['activity_type_id.grouping'] = category;
+
+      urlParams = urlParams || {};
+      urlParams.activityId = urlParams.activityId || 0;
+
+      if (urlParams.category) {
+        activityFilters['activity_type_id.grouping'] = urlParams.category;
       }
-      if (statusType) {
-        af.status_id = CRM.civicase.activityStatusTypes[statusType];
+
+      if (urlParams.statusType) {
+        activityFilters.status_id = CRM.civicase.activityStatusTypes[urlParams.statusType];
       }
-      if (status) {
-        af.status_id = [_.findKey(CRM.civicase.activityStatuses, function (statusObj) {
-          return statusObj.name === status;
+
+      if (urlParams.status) {
+        activityFilters.status_id = [_.findKey(CRM.civicase.activityStatuses, function (statusObj) {
+          return statusObj.name === urlParams.status;
         })];
       }
-      var p = {
-        caseId: caseId,
-        tab: 'activities',
-        aid: id || 0,
+
+      activityFilters = angular.extend(
```

Comment:
Not sure what is happening here? as it was not present before.

Reply:
these are the `af` params passed to the URL. It will use the params in this order:

1. The current `af` params
2. Some params passed to the `getActivityFeedUrl` service like `category`, `statusType`, `status`, etc. This is done for legacy reasons and also because you can pass other params that are not activity related, like the `caseId`.
3. Some more `activityFilters`/`af` params passed to the `getActivityFeedUrl` service `@moreFilters`, `activity_date_time`, etc.